indexd/src/app_client.rs

```diff
@@ -688,6 +695,20 @@ mod tests {
         client.unpin_slab(&slab_id).await.unwrap();
     }
 
+    #[tokio::test]
+    async fn test_prune_slabs() {
+        let server = Server::run();
+
+        server.expect(
```

Comment:
Make sure the request body is empty too

```suggestion
        server.expect(
            Expectation::matching(all_of![
                request::method_path("POST", "/slabs/prune"),
                request::body(""),
            ])
            .respond_with(Response::builder().status(StatusCode::OK).body("").unwrap()),
        );
```

Reply:
Try this:

```
diff --git a/indexd/src/app_client.rs b/indexd/src/app_client.rs
index da278e3..129f67f 100644
--- a/indexd/src/app_client.rs
+++ b/indexd/src/app_client.rs
@@ -3,7 +3,7 @@ use std::time::Duration;
 use blake2b_simd::Params;
 use chrono::{DateTime, Utc};
 use reqwest::{Method, StatusCode};
-use serde_json::to_vec;
+use serde_json::to_string;
 use sia::encryption::EncryptionKey;
 use sia::rhp::Host;

@@ -328,12 +328,12 @@ impl Client {
         path: &str,
         body: &S,
     ) -> Result<D> {
-        let body = to_vec(body)?;
+        let body = to_string(body)?;
         let url = self.url.join(path)?;
         let query_params = self.sign(
             &url,
             Method::POST,
-            Some(&body),
+            Some(body.as_bytes()),
             Utc::now() + Duration::from_secs(60),
         );
         Self::handle_response(
```

Reply:
```
diff --git a/indexd/src/app_client.rs b/indexd/src/app_client.rs
index da278e3..1a7d56b 100644
--- a/indexd/src/app_client.rs
+++ b/indexd/src/app_client.rs
@@ -3,7 +3,7 @@ use std::time::Duration;
 use blake2b_simd::Params;
 use chrono::{DateTime, Utc};
 use reqwest::{Method, StatusCode};
-use serde_json::to_vec;
+use serde_json::to_string;
 use sia::encryption::EncryptionKey;
 use sia::rhp::Host;
 
@@ -328,12 +328,12 @@ impl Client {
         path: &str,
         body: &S,
     ) -> Result<D> {
-        let body = to_vec(body)?;
+        let body = to_string(body)?;
         let url = self.url.join(path)?;
         let query_params = self.sign(
             &url,
             Method::POST,
-            Some(&body),
+            Some(body.as_bytes()),
             Utc::now() + Duration::from_secs(60),
         );
         Self::handle_response(
@@ -702,7 +702,7 @@ mod tests {
         server.expect(
             Expectation::matching(all_of![
                 request::method_path("POST", "/slabs/prune"),
-                request::body("null"),
+                request::body(""),
             ])
             .respond_with(Response::builder().status(StatusCode::OK).body("").unwrap()),
         );
```


```
failures:
    app_client::tests::test_prune_slabs

test result: FAILED. 19 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 2.60s

error: test failed, to rerun pass `--lib`
```